mgnify_pipelines_toolkit/analysis/assembly/mgnify2mimc_inputs.py

```diff
@@ -0,0 +1,726 @@
+                        "pfam"
+                        in downloadable_file["attributes"]["description"][
+                            "label"
+                        ].lower()
+                    ):
+
+                        # url of pfam file
+                        url = downloadable_file["links"]["self"]
+                        mgya_pfam_df = pd.read_csv(
+                            url,
+                            sep=",",
+                            names=["protein", "PfamID", "pfam_description"],
+                        )
+
+                        mgya_pfams = list(mgya_pfam_df["PfamID"])
+
+                        break
+            else:
+                print(
+                    f"Connection attempt failed with status code: {response.status_code}"
+                )
+        except requests.exceptions.RequestException as e:
+            print(f"Connection attempt {attempt + 1} failed due to error: {e}")
+
+        try:
+            response = requests.get(
+                f"https://www.ebi.ac.uk/metagenomics/api/v1/analyses/{mgya_id}"
+            )
+            if response.status_code == 200:
+                mgya_metadata = response.json()["data"]["relationships"]
+                sample_id = mgya_metadata["sample"]["data"]["id"]
+                assembly_id = mgya_metadata["assembly"]["data"]["id"]
+            else:
+                print(
+                    f"Attempt {attempt + 1} failed with status code: {response.status_code}"
+                )
+
+        except requests.exceptions.RequestException as e:
+            print(
+                f"Attempt {attempt + 1} failed due to error: {e}. Please try again later"
+            )
+
+    return sample_id, assembly_id, mgya_pfams
+
+
+# for a given mgys_id, download list of pfams for each mgya and get pfams for each mgya into a pfam dict
+def download_pfams_from_mgys(mgys_id):
+
+    samples_pfam_dict = {}
+
+    mgya_list = extract_mgya_from_mgys(mgys_id)
+
+    # go through list and download pfams
+    for mgya_id in mgya_list:
+        print(f"extracting pfams for {mgya_id}")
+        sample_id, assembly_id, pfams_list = download_pfam_from_mgya(mgya_id)
+
+        time.sleep(2)
+
+        samples_pfam_dict[mgya_id] = pfams_list
+
+    # print (samples_pfam_dict)
+
+    return samples_pfam_dict
+
+
+def download_pfams_from_mgya_list(mgya_list):
+
+    samples_pfam_dict = {}
+
+    # go through list and download pfams
+    for mgya_id in mgya_list:
+        print(f"extracting pfams for {mgya_id}")
+        sample_id, assembly_id, pfams_list = download_pfam_from_mgya(mgya_id)
+
+        time.sleep(2)
+
+        samples_pfam_dict[mgya_id] = pfams_list
+
+    return samples_pfam_dict
+
+
+def download_pfam_clan(version="32.0"):
+
+    # only proceed if pfam version is in list
+    pfam_versions = [
+        "32.0",
+        "33.0",
+        "33.1",
+        "34.0",
+        "35.0",
+        "36.0",
+        "37.0",
+        "37.1",
+        "37.2",
+    ]
```

Comment:
add those to constants again.. I'm not sure as well why those versions? add comment to code why those not only latest?